platform/mds/mds-test-bundle/src/test/java/org/motechproject/mds/test/osgi/MdsDdeBundleIT.java

```diff
@@ -470,6 +488,51 @@ public void testSetOfEnumsAndSetOfStrings() {
         assertTrue(CollectionUtils.contains(retrieved.getRecipients().iterator(), "Christina"));
     }
 
+    @Test 
+    public void testCascadeDelete2() {
+        EntityX x1 = new EntityX();
+        EntityX x2 = new EntityX();
+        
+        EntityZ z1 = new EntityZ();
+        EntityZ z2 = new EntityZ();
+        
+        EntityY y1 = new EntityY();
+        EntityY y2 = new EntityY();
+        
+        x1.getY().add(y1);
+        x1.getY().add(y2);
+        y2.getX().add(x1);
+        y2.getZ().add(z2);
+        y1.getX().add(x1);
+        y1.getZ().add(z1);
+        
+        entityXDataService.create(x1);
+        entityYDataService.create(y1);
+        entityYDataService.create(y2);
+        entityZDataService.create(z1);
+        entityZDataService.create(z2);
+        
+        EntityX testX = entityXDataService.findById(x1.getId());
+        assertNotNull(testX);
+        
+        EntityY testY = entityYDataService.findById(y1.getId());
```

Comment:
Unnecessary. We should rather assert that x instance has got its y relationships set and that the y instances from x have got their z relationships set.